test/verify/check-testlib

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+import subprocess
+
+import parent
+from testlib import *
+
+
+class TestTestLib(unittest.TestCase):
+
+    def testTestCategorization(self):
+        def compareOutput(cmd, *args):
+            output = subprocess.check_output(cmd).strip().split(b"\n\n")
+            self.assertEqual(len(output), len(args))
+            for i in range(0, len(output)):
+                suite = output[i]
+                for test in suite.split(b"\n"):
+                    self.assertIn(test.decode('utf-8'), args[i])
```

Comment:
Is that because the output does not have a predictable order? If so, you could simplify this with set comparison, i. e.

    blocks = subprocess.check_output(cmd).strip().decode().split("\n\n")
    self.assertEqual(set(blocks), set(args))

Of course you can drop the second `set()` if you call the functions with sets in the first place, i. e. using `{}` instead of `[]`.

Reply:
It depends on the order the (Persistent)MachineCase classes get listed by either `inspect.getmembers` or the order in which the tests are given as options to `check-*` or `test/verify/run-test`.

The first should be ordered by name, the second depends on the input.